app/controllers/api/v1/users_controller.rb

```diff
@@ -35,17 +35,11 @@ def destroy
 
   rescue_from ActiveRecord::RecordNotFound do |exception|
     # TODO: Add error message
-    # If current_user raise this exception for delete or update, it means that the user is not logged in
-    # (and therefore not authorized to access this resource)
-    if request.delete? || request.put?
-      render_error(4010, exception.message)
-    else
-      render_error(4000, exception.message)
-    end
+    render_error(4100, exception.message)
   end
 
   rescue_from AuthenticationError do
-    render_error(4011)
+    render_error(4002)
```

Comment:
Why are using here the same error code than in the products controller? i think we can use a new error code here and make errors descriptions more specific